pages/about/AboutContextHandler.php

```diff
@@ -135,16 +141,85 @@ public function editorialMasthead($args, $request)
     }
 
     /**
-     * Display editorialTeam page.
+     * Display editorial history page.
      *
      * @param array $args
      * @param \PKP\core\PKPRequest $request
+     *
+     * @hook AboutContextHandler::editorialHistory [[$mastheadRoles, $mastheadUsers]]
      */
-    public function editorialTeam($args, $request)
+    public function editorialHistory($args, $request)
     {
+        $context = $request->getContext();
+
+        $savedMastheadUserGroupIdsOrder = (array) $context->getData('mastheadUserGroupIds');
+
+        $collector = Repo::userGroup()->getCollector();
+        $allMastheadUserGroups = $collector
+            ->filterByContextIds([$context->getId()])
+            ->filterByMasthead(true)
+            ->orderBy($collector::ORDERBY_ROLE_ID)
+            ->getMany()
+            ->toArray();
+
+        // sort the masthead roles in their saved order for display
+        $mastheadRoles = array_replace(array_flip($savedMastheadUserGroupIdsOrder), $allMastheadUserGroups);
+
+        $mastheadUsers = [];
+        foreach ($mastheadRoles as $mastheadUserGroup) {
+            if ($mastheadUserGroup->getRoleId() == Role::ROLE_ID_REVIEWER) {
+                continue;
+            }
+            // Get all users that were active and are not active any more in the given role
+            // and that have accepted to be displayed on the masthead for that role.
+            // No need to filter by context ID, because the user groups are already filtered so.
+            $usersCollector = Repo::user()->getCollector();
+            $users = $usersCollector
```

Comment:
Running this query inside a loop could be quite expensive on a large database. I can think of a few approaches...

1. Cache the data structures that you're building in this function using a file cache or similiar.
2. Move this query outside the loop, and have it use `whereIn(...)` to fetch relevant results for all the user groups.

Both might be a good combination!